include/openthread/unstable/p2p.h

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2025, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the OpenThread unstable P2P (peer-to-peer) API.
+ */
+
+#ifndef OPENTHREAD_UNSTABLE_P2P_H_
+#define OPENTHREAD_UNSTABLE_P2P_H_
+
+#include <openthread/unstable/link.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @addtogroup unstable-api-p2p
+ *
+ * @brief
+ *   This module includes unstable functions for the Thread P2P link.
+ *
+ * @note
+ *   The functions in this module require `OPENTHREAD_CONFIG_PEER_TO_PEER_ENABLE=1`.
+ *
+ * @{
+ */
+
+/**
+ * Represents a request for waking up the peer to establish P2P links.
+ */
+typedef struct otP2pRequest
+{
+    otWakeupAddress mWakeupAddress; ///< Wake-up address of the peer.
+} otP2pRequest;
+
+/**
+ * Informs the caller about the result of establishing P2P links with peers.
+ *
+ * @param[in] aError   OT_ERROR_NONE    Indicates that at least one P2P link has been successfully established. The
+ *                                      P2P link state changes can be observed by @sa otP2pEventCallback().
+ *                     OT_ERROR_TIMEOUT Indicates that no P2P link has been established before the wake-up window ends.
+ * @param[in] aContext A pointer to application-specific context.
+ */
+typedef void (*otP2pLinkedCallback)(otError aError, void *aContext);
```

Comment:
Is this callback needed given we have  the otP2pEventCallback?

Reply:
The `otP2pLinkedCallback ` is mainly used to notify the caller that the P2P link process has been finished.  Then the caller can start the next P2P link operations. The `otP2pEventCallback` only notify the caller that the link has been established or torn down.  Especially when the P2P link fails, the `otP2pEventCallback` won't be called.